### PR TITLE
Load categories from config and improve app sorting

### PR DESCRIPTION
**Pull request by maxaleks**
Resolves https://github.com/blockscout/frontend/issues/1498

1. added a new ENV: `NEXT_PUBLIC_MARKETPLACE_CATEGORIES_URL`
2. made loading categories from a separate config file
3. improved app sorting

### CODE REVIEW

**Review by tom2drum — CHANGES_REQUESTED** [review]

File: ui/marketplace/useMarketplaceCategories.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const feature = config.features.marketplaceCategories;
+const configUrl = feature.isEnabled ? feature.configUrl : '';
+
+export default function useMarketplaceCategories() {
+  const apiFetch = useApiFetch();
+  const { isPlaceholderData, isError, error, data } = useQuery<unknown, ResourceError<unknown>, Array<string>>({
+    queryKey: [ 'marketplace-categories' ],
+    queryFn: async() => apiFetch(configUrl, undefined, { resource: 'marketplace-categories' }),
+    select: (data) => (data as Array<string>),
```
[the remark this answers is not in the document]
Comment (maxaleks) [reply ui/marketplace/useMarketplaceCategories.tsx:16]: I don't think so. I was just following an example and decided not to delete anything that might be necessary

File: ui/marketplace/useMarketplace.tsx [hunk 2]
```diff
@@ -73,14 +75,23 @@ export default function useMarketplace() {
 
   const { isPlaceholderData, isError, error, data, displayedApps } = useMarketplaceApps(debouncedFilterQuery, selectedCategoryId, favoriteApps);
 
+  const { value: isExperiment } = useFeatureValue('marketplace_exp', false);
+  const { isPlaceholderData: isCategoriesPlaceholderData, data: categoriesData } = useMarketplaceCategories();
+
   const categories = React.useMemo(() => {
+    let categoryNames: Array<string> = [];
```
[the remark this answers is not in the document]
Comment (maxaleks) [reply ui/marketplace/useMarketplace.tsx:82]: > What if there is a fetch error? Or if the category config is not provided?

I think `categoriesData` would then be empty, wouldn't it?
https://github.com/blockscout/frontend/pull/1545/files#diff-2ebed5c8ee6bd0ee84fe6b654067cdee18307a229868afd84e6652d64a866517R85

**Comment by tom2drum** [comment]
Can we make the tab skeleton resemble real tabs more? For instance, by adding additional tabs and increasing their width.

image

File: ui/marketplace/useMarketplace.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tom2drum) [reply ui/marketplace/useMarketplace.tsx]: Oh yeah, you're right. I missed the condition for the placeholder data in the query hook.

**Review by tom2drum — APPROVED** [review]

**Pull request merged**